build.py

```diff
@@ -811,6 +820,16 @@ def write_rules(self, ninja):
                     rspfile_content = self.tool_commands['LINK'].replace('$LINK ', ''),
                     pool='winlink',
                     description = 'LINK $out')
+            if 'SHLINK' in self.tool_commands:
+                if 'LINK' not in self.tool_commands:
+                    ninja.pool('winlink', GetOption('link-pool-depth'))
```

Comment:
Now that this is shared, can you remove it from LINK/SHLINK and pull it up to the top of the windows tool block (line ~291).